src/analytics/private/scan-metadata-workflow.ts

```diff
@@ -50,36 +51,29 @@ export class ScanMetadataWorkflow extends Construct {
   constructor(scope: Construct, id: string, props: ScanMetadataWorkflowProps) {
     super(scope, id);
 
-    // create Step function workflow to orchestrate the workflow to scan metadata.
-    this.scanMetadataWorkflow = this.createWorkflow(props);
-
-    // Create an EventBridge Rule to trigger the workflow periodically
-    const rule = new Rule(scope, 'ScanMetadataScheduleRule', {
-      // schedule: Schedule.expression('cron(0 1 * * ? *)'),
-      schedule: Schedule.expression(props.scanMetadataWorkflowData.scheduleExpression),
+    // create DynamoDB table to store workflow info
+    const workflowInfoTable = new Table(this, 'MetadataWorkflowInfoTable', {
```

Comment:
We are introducing too many tables. Reusing the existing table in this module or save the data to S3.

Reply:
Stored workflow job info into S3 files.